__tests__/app.test.js

```diff
@@ -291,10 +291,59 @@ describe("app.js", () => {
                 author: "icellusedkars",
                 article_id: 1,
                 created_at: expect.any(String),
+                comment_id: 19,
               })
             );
           });
       });
+      test("status:400, responds with invalid input with post attempt to an invalid article_id", () => {
+        return request(app)
+          .post("/api/articles/xzibit/comments")
+          .send({
+            username: "icellusedkars",
+            body: "Checkmate... Did he just throw my cat out of the window? You're a very talented young man, with your own clever thoughts and ideas. Do you need a manager? I gave it a cold? I gave it a virus. A computer virus. Yeah, but John, if The Pirates of the Caribbean breaks down, the pirates don’t eat the tourists.",
+          })
+          .expect(400)
+          .then(({ body: { msg } }) => {
+            expect(msg).toBe("Invalid input");
+          });
+      });
+      test("status:400, responds with invalid post input when passed a post body with an invalid / missing parameter", () => {
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send({
+            usario: "icellusedkars",
+            body: "Jaguar shark! So tell me - does it really exist? So you two dig up, dig up dinosaurs? You really think you can fly that thing? Life finds a way. Hey, you know how I'm, like, always trying to save the planet? Here's my chance. You know what? It is beets. I've crashed into a beet truck.",
+          })
+          .expect(400)
+          .then(({ body: { msg } }) => {
+            expect(msg).toBe("Invalid post input");
+          });
+      });
+      test("status:400, responds with invalid post data type when passed a post body with incorrect parameter data type", () => {
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send({
+            username: 1,
```

Comment:
This one's a little tricky as by the time it's injected into the SQL query, it will be a string. Think of it as numbers being perfectly valid inside of strings, but NOT the other way around.
This test is _**really**_ a test for the username NOT being present in the users table